frappe/model/base_document.py

```diff
@@ -384,6 +385,10 @@ def get_valid_dict(
 			if ignore_nulls and not is_virtual_field and value is None:
 				continue
 
+			# If the docfield is not nullable - set a default non-null value
+			if value is None and getattr(df, "not_nullable", False):
+				value = get_not_null_defaults(df.fieldtype)
```

Comment:
What is doctype has non-empty string default? 


end goal: `frappe.new_doc()` should always have the defaults set.